### PR TITLE
Fix/41 48 download dataset preview

### LINKED ISSUE #41: readr is not installed error
## Describe the bug
readr is a dependency for

## How to reproduce
Steps to reproduce the behaviour:
1. install eesyapi in a fresh project
2. try to use the `download_dataset()` function

## Expected behaviour
It just works

## Screenshots
image

## Additional context
Had initially left readr as only suggested, as just adding readr as an import gives us warning as package maintainers that we're calling an import that isn't used (we call httr, which then depends on readr, but for some reason doesn't seem to install it).

### LINKED ISSUE #48: Give specific example of using download dataset as a preview function
## Is your feature request related to a problem? Please describe.
Had a suggestion of a `preview_dataset()` function that is effectively 

`download_dataset("id....") |> head()` 

## Describe the solution you'd like
Quickest option will be to make it clearer in the download_dataset section of the main workflow vignette that it should be seen as a preview function, and encourage the piping into head() to limit the rows in memory.

## Describe alternatives you've considered
Considering making a quick `preview_dataset()` function, though thought that might be overkill for now. Especially as currently we can't do anything other than pull in the whole data set.

## Additional context
None.

### PR DESCRIPTION
**Pull request by cjrace**
# Brief overview of changes

Repurposed `dataset_preview()` into a new function `preview_dataset()`. 

## Why are these changes being made?

Mostly in response to the two issues raised from initial testing today #48 and #41. I went down a path of trying to use httr2 over httr and then realised how I could make use of readr if I called it myself with the n_max argument, doing exactly what we wanted and had been considering using the GET or POST endpoints for.

Changing the naming from download to preview also makes the function purpose much clearer in the workflow for users and feels less jarring than the option to download against option to query (also as it technically was reading not downloading).

## Detailed description of changes

- `download_dataset()` has been renamed to `preview_dataset()`
- Added a new argument of `n_max = ` to `preview_dataset()`
- Documentation and main workflow vignette have been updated to reflect the preview nature of the function.
- Added a new utils helper to check if a value is an integer (SO suggestion as not available in base R)
- Explicitly used readr to get the CSV data, and switched to using httr2 over httr
- Updated examples in `preview_dataset()` documentation and added tests to cover the new behaviour
- Specified the language in the DESCRIPTION as I realised we weren't doing this yet and probably should

## Additional information for reviewers

As it's the newer and improved package / method, we should generally shift over to httr2 where we can over httr.

I've bumped to a dev version and assuming we'll keep on that for mop up and then do another minor version bump to v0.4.0 to track what version we're using in user testing.

I'll admit I don't understand exactly how `read_csv()` can only request x rows as I didn't think that was possible from the endpoint, but I've done some benchmarking using the big attendance example on dev and it certainly seems to do something to avoid reading everything into memory as it's much faster getting 10 rows a time than the full data set...

New `preview_dataset()` function with 10 rows as default averages 3.66 seconds (this even includes a quick pre-query to check we can get a successful response):
image

Old `download_dataset()` function on main branch getting all rows averages 18.3 seconds:
image

## Issue ticket number/s and link

Resolves #48 by making it a preview function, and fixes #41.

### CODE REVIEW

File: vignettes/ees-api-workflow.Rmd [hunk 1]
```diff
@@ -174,23 +174,28 @@ eesyapi::get_meta(dataset_id = "7c0e9201-c7c0-ff73-bee4-304e731ec0e6") |>
...
 
-### Downloading an entire data set in a single file
+### Preview raw CSV data
 
-Whilst specific queries (as detailed in the following sections) are best used for automated pipeline 
-workflows, the EES API also offers the option
-to just download the entirety of the underlying data file for a given data set. This bypasses the 
-use of IDs, making it a simpler short-term option to retrieve data, but with the risk of creating
-breaking changes in automated pipelines should element labels change.
+In the EES API you have the option to preview underlying data CSV file for a given data set.
```
Comment (rmbielby) [inline vignettes/ees-api-workflow.Rmd:179]: "preview underlying data" -> "preview the underlying data".
Also wondering whether the end user needs to know / care that it's a csv file behind this? Might just be extra info that they don't need to know here.

File: vignettes/ees-api-workflow.Rmd [hunk 2]
```diff
@@ -174,23 +174,28 @@ eesyapi::get_meta(dataset_id = "7c0e9201-c7c0-ff73-bee4-304e731ec0e6") |>
 
 ## Retrieving data from a data set
 
-### Downloading an entire data set in a single file
+### Preview raw CSV data
```
Comment (rmbielby) [inline vignettes/ees-api-workflow.Rmd:177]: Makes sense here thematically, but just wondering if actually in terms of a real user's journey, whether the data preview bit would be better being put before the meta data query? Feel like that was part of what they were getting at in the session yesterday.

File: vignettes/ees-api-workflow.Rmd [hunk 3]
```diff
@@ -174,23 +174,28 @@ eesyapi::get_meta(dataset_id = "7c0e9201-c7c0-ff73-bee4-304e731ec0e6") |>
...
-This will return a data frame containing the data in the form that DfE analysts uploaded it to 
-EES in.
+This will return a data frame containing the first 10 rows of the data in the form that DfE 
+analysts uploaded it to EES in.
+
+While you can set n_max to Inf, and get the whole data set using this, you should avoid relying on
+the `preview_dataset()` function in your pipelines or dashboards and instead use it to explore the
+data as you create your own custom query using `query_dataset()`.
+
+If you only read the whole CSV, you risk:
```
Comment (rmbielby) [inline vignettes/ees-api-workflow.Rmd:196]: The bullet list isn't formatting properly, probably needs a blank line before the first bullet.

**Review by rmbielby — CHANGES_REQUESTED** [review]

File: vignettes/ees-api-workflow.Rmd (hunk 1 above)
Comment (cjrace) [reply vignettes/ees-api-workflow.Rmd]: Makes sense, updated in PR response commit

File: vignettes/ees-api-workflow.Rmd (hunk 2 above)
Comment (cjrace) [reply vignettes/ees-api-workflow.Rmd]: Moved upwards in PR response commit

File: vignettes/ees-api-workflow.Rmd (hunk 3 above)
Comment (cjrace) [reply vignettes/ees-api-workflow.Rmd]: Good spot, fixed in PR response commit

**Review by rmbielby — APPROVED** [review]

**Pull request merged**